CAP/gap/PrintingFunctions.gi

```diff
@@ -3,133 +3,40 @@
-        
-        for j in Reversed( caps_positions ) do
-            current := Concatenation( current{[ 1 .. j - 1 ]}, " ", current{[ j .. Length( current ) ]} );
-        od;
-        
-        Add( name_list, [ i, current ] );
-        
-    od;
-    
-    return_list := [ ];
-    
-    for i in name_list do
-        
-        entries := CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( i [ 1 ] );
-        
-        added := false;
-        
-        for j in return_list do
-            
-            if ForAll( j, k -> IsSubset( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( k[ 1 ] ), entries )
-                            or IsSubset( entries, CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( k[ 1 ] ) ) ) then
-                
-                Add( j, i );
-                added := true;
-                break;
-                
-            fi;
-            
-        od;
-        
-        if not added then
-            
-            Add( return_list, [ i ] );
-            
-        fi;
-        
-    od;
-    
-    for i in return_list do
-        Sort( i, function( j, k ) return IsSubset( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( j[ 1 ] ), CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD.( k[ 1 ] ) ); end );
-    od;
-    
-    # Some cosmetics for the lists:
-    # Write Abel with a capital, its the man ;).
-    # Change the order so that the list with Abelian is at the end.
-    
-    for i in return_list do
-        for j in i do
-            j[ 2 ] := ReplacedString( j[ 2 ], "abelian", "Abelian" );
-        od;
-    od;
-    
-    for i in [ 1 .. Length( return_list ) ] do
-        if ForAny( return_list[ i ], j -> j[2] = "Abelian" ) then
-            return_list := Concatenation( return_list{[ 1 .. i - 1 ]}, return_list{[ i + 1 .. Length( return_list ) ]}, [ return_list[ i ] ] );
-        fi;
-    od;
-    
-    return return_list;
-    
-end );
-
-CAP_INTERNAL_CATEGORY_PROPERTY_RANK_AND_STRING.sorted_list := __CAP_INTERNAL_RANK_CATEGORY_FROM_METHOD_RECORD();
-
 InstallGlobalFunction( InfoStringOfInstalledOperationsOfCategory,
   
   function( category )
-    local prim_operations, derived_operations, adjectives, current_stack, current_property, current_adjective, string;
+    local prim_operations, derived_operations, list_of_properties, property, string;
     
     if not IsCapCategory( category ) then
         Error( "first argument must be a category" );
         return;
     fi;
     
-    prim_operations := ListPrimitivelyInstalledOperationsOfCategory( category );
-    derived_operations := ListInstalledOperationsOfCategory( category );
+    prim_operations := Length( ListPrimitivelyInstalledOperationsOfCategory( category ) );
+    derived_operations := Length( ListInstalledOperationsOfCategory( category ) );
     
-    prim_operations := Length( prim_operations );
-    derived_operations := Length( derived_operations );
+    list_of_properties := ShallowCopy( RecNames( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD ) );
```

Comment:
```suggestion
    list_of_properties := Intersection( ListKnownCategoricalProperties( category ), RecNames( CAP_INTERNAL_CONSTRUCTIVE_CATEGORIES_RECORD ) );
```

and get rid of the next line. (Strictly speaking, this is a bugfix independent of this PR, but I think it makes sense to include it here.)

The issue: Currently the displayed properties are chosen simply by looking at the installed operations. But CAP cannot distinguish for example between a pre-Abelian and an Abelian category simply by looking at the operations. So I think we should always restrict to known categorical properties.

Reply:
This makes `IsEquippedWithHomomorphismStructure` disappear since it is not set by default. I will reset to the first behavior and we can take care of setting `IsEquippedWithHomomorphismStructure` in a later PR.

Reply:
The idea is that I want to show what can be computed, regardless of whether the property is set or not. Of Course, we can enforce setting the property if all defining operations are available.

Reply:
The problem is that `IsAbelianCategory` cannot be checked in this way (and setting the property forcefully would be wrong in general), because there is no operation `IsomorphismFromCoimageToImage` or some equivalent operation which corresponds to being Abelian.